v2/options.go

```diff
@@ -42,6 +66,33 @@ type ReadWriteOption interface {
 	readWriteOption()
 }
 
+// SplitReadWriteOptions splits the rwopts by type into ReadOption and WriteOption slices.
+func SplitReadWriteOptions(rwopts ...ReadWriteOption) ([]ReadOption, []WriteOption) {
+	var ropts []ReadOption
+	var wopts []WriteOption
+	for _, opt := range rwopts {
+		switch opt := opt.(type) {
+		case ReadOption:
+			ropts = append(ropts, opt)
+		case WriteOption:
+			wopts = append(wopts, opt)
+		}
+	}
+	return ropts, wopts
+}
+
+// MergeReadWriteOptions merges given ro and wo options into a single slice of ReadWriteOption.
+func MergeReadWriteOptions(ro []ReadOption, wo []WriteOption) []ReadWriteOption {
+	var rwopts []ReadWriteOption
+	for _, o := range ro {
+		rwopts = append(rwopts, o)
+	}
+	for _, o := range wo {
+		rwopts = append(rwopts, o)
+	}
+	return rwopts
+}
```

Comment:
hmm, perhaps this is overkill/unnecessary - you never actually merge them. you just convert from one slice type to another. that's three lines of very trivial Go, so I don't think it warrants new API.